File: Java/프로그래머스/3/12927. 야근 지수/야근 지수.java
import java.util.*;

class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int work : works) {
            pq.offer(work);
        }
        while (n > 0) {
            int temp = pq.poll();
            if (temp == 0) break;
            temp--;
            n--;
            pq.offer(temp);
        }
        while (!pq.isEmpty()) {
            int temp = pq.poll();
            answer += (long) temp * temp;
        }
        return answer;
    }
}

File: Python/프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq


def solution(n, works):
    answer = 0

    if sum(works) < n:
        return answer

    q = []
    for work in works:
        heapq.heappush(q, -work)

    while n > 0:
        m = heapq.heappop(q)
        if m == 0:
            break
        heapq.heappush(q, m + 1)
        n -= 1

    for i in q:
        answer += i**2

    return answer